assets/model_monitoring/components/src/shared_utilities/store_url.py

```diff
@@ -186,12 +215,23 @@ def write_file(self, file_content: Union[str, bytes], relative_path: str = None,
 
         container_client = self.get_container_client(credential)
         full_path = f"{self.path}/{relative_path.strip('/')}" if relative_path else self.path
-        if isinstance(container_client, FileSystemClient):
-            with container_client.get_file_client(full_path) as file_client:
-                return file_client.upload_data(file_content, overwrite)
-        else:
-            with container_client.get_blob_client(full_path) as blob_client:
-                return blob_client.upload_blob(file_content, overwrite=overwrite)
+        # TODO: edit this check block after we are able to support submitting managed identity MoMo graphs.
+        try:
+            if isinstance(container_client, FileSystemClient):
+                with container_client.get_file_client(full_path) as file_client:
+                    return file_client.upload_data(file_content, overwrite)
+            else:
+                with container_client.get_blob_client(full_path) as blob_client:
+                    return blob_client.upload_blob(file_content, overwrite=overwrite)
+        except CredentialUnavailableError as cue:
+            if "AzureML Spark On Behalf of credentials not available in this environment" in cue.message:
+                raise InvalidInputError(
+                    f"Failed to use AML OBO token for data read."
```

Comment:
for data write?

Reply:
Will add constant and update the message